tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -231,11 +217,38 @@ internal void WriteClosedSpans()
 
             if (spansToWrite.Count > 0)
             {
+                _ = GetSamplingPriority(TriggerSamplingDecision.IfNotSet);
+
                 RunSpanSampler(spansToWrite);
                 Tracer.Write(spansToWrite);
             }
         }
 
+        /// <summary>
+        /// Gets the trace's sampling priority, optionally triggering a sampling decision.
+        /// </summary>
+        public int? GetSamplingPriority(TriggerSamplingDecision trigger = TriggerSamplingDecision.None)
```

Comment:
I wonder about this API 🤔 given basically everywhere uses IfNotSet, I wonder if we should just make this a required arg instead of optional so it's very obvious the behaviour. 

Also, if it's `IfNotSet` (or `Always`), then it always returns a non-null value, right? In which case maybe it actually makes sense to have two different APIs: one that makes a sampling decision (or overwrites it) and returns `int`, one which gets the current without making a decision and returns `int?`. We can update the docs for the methods to make it clear _when_ to use each method then? 🤔  Not sure, WDYT?

Reply:
I changed this to use a bool instead of an enum, since we never use `Always` to force multiple sampling decision on the same trace. `bool triggerSamplingDecision = false`

We do have many calls that use `None`/`false`, that's why I left that as a default, but happy to make it not optional.

> Also, if it's `IfNotSet` (or `Always`), then it always returns a non-null value, right?

Not necessarily. If we try to make a sampling decision but there are no spans yet in the `TraceContext`, we don't make the sampling decision and it remains null. We can't make a sampling decision without a trace id, and currently trace ids live in the spans (or more accurately, in the span context, for some reason), not in the trace context as (I think) it should.

Reply:
I ended up going with your suggestion of having two APIs after all. Thanks!